src/Microsoft.Azure.ServiceBus.MessageId/MessageIdPlugin.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Microsoft.Azure.ServiceBus.MessageId
+{
+    /// <summary>Generate Azure Service Bus <see cref="Message"/> id.</summary>
+    public class MessageIdPlugin : ServiceBusPlugin
+    {
+        private readonly Func<string> messageIdGenerator;
```

Comment:
@SeanFeldman , We would be just exposing as a get property. I would raise the same argument for exposing GetPlugins() on a receiver/queueClient. We already know what we are initializing it with. 
Right now there's no way to distinguish one MessagIdPlugin with another. There's no way to identify the right plugin. I don't think this is a necessity, but it would be nice to know what the plugin is initialized with.